core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -135,7 +137,15 @@ public CloseableIterable<FileScanTask> planFiles() {
             planningDuration.stop();
             ScanReport scanReport =
                 ImmutableScanReport.builder()
-                    .projection(schema())
+                    .schemaId(schema().schemaId())
+                    .projectedFieldIds(
+                        schema().columns().stream()
+                            .map(NestedField::fieldId)
+                            .collect(Collectors.toList()))
+                    .projectedFieldNames(
+                        schema().columns().stream()
+                            .map(NestedField::name)
+                            .collect(Collectors.toList()))
```

Comment:
This should use the set of projected IDs and look up names using `Schema.findColumnName(Integer)`.

Reply:
done